File: deposito/DefaultComboBoxModelAndRenderer.java
package javascript.swing.MnR;

/**
 * The default implementation of the AbstractComboBoxModelAndRenderer
 *
 * @author gianpiero.diblasi
 * @param <T> The type
 */
public class DefaultComboBoxModelAndRenderer<T> extends AbstractComboBoxModelAndRenderer<T> {

  @Override
  protected String render(T element) {
    return element.toString();
  }
}

File: transpile/javascript/swing/MnR/DefaultComboBoxModelAndRenderer.js
/**
 * The default implementation of the AbstractComboBoxModelAndRenderer
 *
 * @author gianpiero.diblasi
 * @param <T> The type
 */
class DefaultComboBoxModelAndRenderer extends AbstractComboBoxModelAndRenderer {

   render(element, inlist) {
    let label = document.createElement("label");
    label.textContent = element ? element.toString() : "";
    return label;
  }
}
